File: leetcode/string/266. Palindrome Permutation/py/main.py
class Solution:
    # Time O(N)
    # Space O(1)
    def canPermutePalindrome(self, s: str) -> bool:
        counter = Counter(s)
        cnt_odd = 0

        for k in counter:
            if counter[k] % 2 != 0:
                if cnt_odd > 0: return False
                cnt_odd += 1
        return True

File: leetcode/string/266. Palindrome Permutation/js/main.js
// Time O(N)
// Space O(1)
const canPermutePalindrome = str => {
  let map = {};

  for (let s of str) {
    map[s] = ~~map[s] + 1;
  }

  let cnt = 0;

  for (let key of Object.keys(map)) {
    if (cnt > 1) {
      return false;
    }

    if (map[key] % 2 !== 0) {
      cnt++;
    }
  }

  return cnt <= 1;
};
